src/solvers/nlopt_optimization_solver.C

```diff
@@ -128,7 +127,8 @@ void __libmesh_nlopt_equality_constraints(unsigned m,
   if (sys.solution->size() != n)
     libmesh_error_msg("Error: Input vector x has different length than sys.solution!");
 
-  for (unsigned int i=sys.solution->first_local_index(); i<sys.solution->last_local_index(); i++)
+  for (auto i : IntRange<dof_id_type>(sys.solution->first_local_index(),
+                                      sys.solution->last_local_index()))
```

Comment:
Should we add an index_range overload that takes NumericVector?

Reply:
Sure. For that one I'd probably hard code `numeric_index_type`? That seems to be what we're using with `NumericVectors`...

Reply:
Right.